libs/core-ui/src/lib/Cohort/CohortInfo/CohortInfo.tsx

```diff
@@ -15,131 +15,84 @@ import { cohortInfoStyles } from "./CohortInfo.styles";
-                    {localization.ErrorAnalysis.CohortInfo.correct}
-                  </div>
-                  <div className={classNames.tableData}>
-                    {this.props.currentCohort.cohortStats.totalCohortCorrect}
-                  </div>
-                </Stack>
-              )}
-              {this.props.currentCohort.cohortStats instanceof
-                ErrorCohortStats && (
-                <Stack horizontal tokens={alignmentStackTokens}>
-                  <div className={classNames.tableData}>
-                    {localization.ErrorAnalysis.CohortInfo.incorrect}
-                  </div>
-                  <div className={classNames.tableData}>
-                    {this.props.currentCohort.cohortStats.totalCohortIncorrect}
-                  </div>
-                </Stack>
-              )}
-            </Stack>
-          </div>
-        </div>
-        {this.props.includeDividers && <div className={classNames.divider} />}{" "}
-        <div className={classNames.section}>
-          <div className={classNames.header}>
-            {localization.ErrorAnalysis.CohortInfo.predictionPath}
-          </div>
+      <Stack className={classNames.container} tokens={{ childrenGap: "20px" }}>
+        <DefaultButton
+          className={classNames.button}
+          text={localization.ErrorAnalysis.CohortInfo.saveCohort}
+          onClick={(): any => this.props.onSaveCohortClick()}
+          disabled={this.props.disabledView}
+        />
+        <Stack>
+          <Label>
+            {localization.ErrorAnalysis.CohortInfo.basicInformation}
+          </Label>
+          {this.props.currentCohort.cohort.name !==
+            localization.ErrorAnalysis.Cohort.defaultLabel && (
+            <Text>{this.props.currentCohort.cohort.name}</Text>
+          )}
+          <Text>
+            {localization.ErrorAnalysis.Cohort.defaultLabel} (
+            {getCohortFilterCount(this.props.currentCohort.cohort)}{" "}
+            {localization.ErrorAnalysis.CohortInfo.filters})
+          </Text>
+        </Stack>
+
+        <Stack>
+          <Label>
+            {localization.ErrorAnalysis.CohortInfo.baseCohortInstances}
+          </Label>
+          <Text>
+            {localization.ErrorAnalysis.CohortInfo.total}{" "}
+            {this.props.currentCohort.cohortStats.totalAll}
+          </Text>
+          {this.props.currentCohort.cohortStats instanceof ErrorCohortStats && (
+            <Text>
+              {localization.ErrorAnalysis.CohortInfo.correct}{" "}
+              {this.props.currentCohort.cohortStats.totalCorrect}
+            </Text>
+          )}
+          {this.props.currentCohort.cohortStats instanceof ErrorCohortStats && (
+            <Text>
+              {localization.ErrorAnalysis.CohortInfo.incorrect}{" "}
+              {this.props.currentCohort.cohortStats.totalIncorrect}
+            </Text>
+          )}
+        </Stack>
+
+        <Stack>
```

Comment:
I think these two stacks could be refactored more into a function with different parameters?  And the text sections could also be a small function?  That might reduce the duplicate code even more.

Reply:
I'm sure! Sadly, I don't really have time to tackle this further at the moment. There are a few other files with similar custom code that we could also fix... I won't do it in this PR, though. As and when I have time 😉